log/logger.go

```diff
@@ -177,19 +177,22 @@ type logger struct {
 }
 
 func (l *logger) write(msg string, lvl Lvl, ctx []interface{}, skip int) {
-	l.h.Log(&Record{
+	record := &Record{
 		Time: time.Now(),
 		Lvl:  lvl,
 		Msg:  msg,
 		Ctx:  newContext(l.ctx, ctx),
-		Call: stack.Caller(skip),
 		KeyNames: RecordKeyNames{
 			Time: timeKey,
 			Msg:  msgKey,
 			Lvl:  lvlKey,
 			Ctx:  ctxKey,
 		},
-	})
+	}
+	if locationEnabled.Load() {
+		record.Call = stack.Caller(skip)
```

Comment:
It's problematic. e.g. in the `log/handler_glog.go`, L188, we assume this is available.

Reply:
Hm, interesting. I dug into this, it turns out that `r.Call.String()` == `%!v(NOFUNC)`. 
due to 
```golang
// String implements fmt.Stinger. It is equivalent to fmt.Sprintf("%v", c).
func (c Call) String() string {
	return fmt.Sprint(c)
}
```
So there's no crash, just it won't print the backtrace.

Reply:
Should be noted, though: the `Call` is `stack.Call` -- not a pointer. Nothing is `nil` here. And it goes on
```
type Call struct {
	frame runtime.Frame
}
```
(no pointers)
```
type Frame struct {
	// PC is the program counter for the location in this frame.
	// For a frame that calls another frame, this will be the
	// program counter of a call instruction. Because of inlining,
	// multiple frames may have the same PC value, but different
	// symbolic information.
	PC uintptr
...
```
This struct too is (nearly) pointer-free.

Reply:
Ha I spent two hours yesterday trying to create a test to hit this, but couldn't because no pointer. So it looks good to me